lib/src/local_working_copy.rs

```diff
@@ -2516,11 +2513,16 @@ impl LockedWorkingCopy for LockedLocalWorkingCopy {
                 })?;
         }
         if self.old_operation_id != operation_id || self.new_workspace_name.is_some() {
-            if let Some(new_name) = self.new_workspace_name {
-                self.wc.checkout_state_mut().workspace_name = new_name;
-            }
-            self.wc.checkout_state_mut().operation_id = operation_id;
-            self.wc.save();
+            let workspace_name = self.new_workspace_name.unwrap_or_else(
+                // Only read the filesystem if we're missing the workspace name.
+                || CheckoutState::load(&self.wc.state_path).workspace_name,
```

Comment:
nit: use `self.wc.workspace_name()`?

I know `self.wc.checkout_state` wouldn't likely be set here, but we'll have to make it loaded eagerly to get rid of bad `.unwrap()`s.